src/utils/clmfates_interfaceMod.F90

```diff
@@ -2593,6 +2643,116 @@ end subroutine wrap_canopy_radiation
 
  ! ======================================================================================
 
+ subroutine WrapSeedGlobal(this,is_restart_flag)
+
+   ! Call mpi procedure to provide the global seed output distribution array to every gridcell.
+   ! This could be conducted with a more sophisticated halo-type structure or distributed graph.
+
+   use decompMod,              only : procinfo
+   use spmdMod,                only : MPI_REAL8, MPI_SUM, mpicom
+   use FatesDispersalMod,      only : lneighbors, neighbor_type, dispersal_type
+   use FatesInterfaceTypesMod, only : numpft_fates => numpft
+
+   ! Arguments
+   class(hlm_fates_interface_type), intent(inout) :: this
+   logical, optional                              :: is_restart_flag 
+
+   ! Local
+   integer :: numg ! total number of gridcells across all processors
+   integer :: ier  ! error code
+   integer :: g    ! gridcell index
+
+   logical :: set_flag ! local logical variable to pass to IsItDispersalTime
+
+   type (neighbor_type),  pointer :: neighbor
+
+   ! If WrapSeedGlobal is being called at the end a fates restart call,
+   ! pass .false. to the set_dispersed_flag to avoid updating the 
+   ! global dispersal date
+   set_flag = .true.
+   if (present(is_restart_flag)) then
+      if (is_restart_flag) set_flag = .false.   
+   end if
+
+   call t_startf('fates-seed-mpi_reduce')
+
+   if (IsItDispersalTime(setdispersedflag=set_flag)) then
+
+      ! Re-initialize incoming seed buffer for this time step
+      this%fates_seed%incoming_global(:,:) = 0._r8
+      this%fates_seed%outgoing_global(:,:) = 0._r8
+
+      ! Distribute outgoing seed data from all nodes to all nodes
+      call MPI_Allgatherv(this%fates_seed%outgoing_local, procinfo%ncells*numpft_fates, MPI_REAL8, &
+                          this%fates_seed%outgoing_global, this%fates_seed%ncells_array*numpft_fates, this%fates_seed%begg_array*numpft_fates, &
+                          MPI_REAL8, mpicom, ier)
+      write(iulog,*) 'WSG: MPI_Allgatherv ier: ', ier
+
+      ! zero outgoing local for all gridcells outside threaded region now that we've passed them out
```

Comment:
If calling outside of a threaded region is required should we test if that's the case here?

Reply:
Let's look into this.

Reply:
We might be able to use `[omp_in_parallel](https://www.openmp.org/spec-html/5.0/openmpsu115.html)` here.  I'll give it a shot.